orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/DeployManifestContext.java

```diff
@@ -17,88 +17,45 @@
-      @JsonProperty("trafficManagement") TrafficManagement trafficManagement,
-      @JsonProperty("requiredArtifactIds") List<String> requiredArtifactIds,
-      @JsonProperty("requiredArtifacts") List<BindArtifact> requiredArtifacts) {
-    this.source = source;
-    this.manifestArtifactId = manifestArtifactId;
-    this.manifestArtifact = manifestArtifact;
-    this.manifestArtifactAccount = manifestArtifactAccount;
-    this.skipExpressionEvaluation = skipExpressionEvaluation;
-    this.trafficManagement =
-        Optional.ofNullable(trafficManagement).orElse(new TrafficManagement(false, null));
-    this.requiredArtifactIds = requiredArtifactIds;
-    this.requiredArtifacts = requiredArtifacts;
-  }
-
-  @Getter
-  public static class BindArtifact {
-    @Nullable private final String expectedArtifactId;
-
-    @Nullable private final Artifact artifact;
-
-    public BindArtifact(
-        @JsonProperty("expectedArtifactId") @Nullable String expectedArtifactId,
-        @JsonProperty("artifact") @Nullable Artifact artifact) {
-      this.expectedArtifactId = expectedArtifactId;
-      this.artifact = artifact;
-    }
-  }
+  private TrafficManagement trafficManagement = new TrafficManagement();
 
-  @Getter
+  @Data
   public static class TrafficManagement {
-    private final boolean enabled;
-    private final Options options;
+    private boolean enabled = false;
+    private Options options = new Options();
 
-    public TrafficManagement(
-        @JsonProperty("enabled") Boolean enabled, @JsonProperty("options") Options options) {
-      this.enabled = Optional.ofNullable(enabled).orElse(false);
-      this.options =
-          Optional.ofNullable(options).orElse(new Options(false, Collections.emptyList(), null));
+    @Data
+    public static class Options {
+      private boolean enableTraffic = false;
+      private List<String> services = Collections.emptyList();
+      private ManifestStrategyType strategy = ManifestStrategyType.None;
     }
 
-    @Getter
-    public static class Options {
-      private final boolean enableTraffic;
-      private final List<String> services;
-      private final ManifestStrategyType strategy;
+    public enum ManifestStrategyType {
+      @JsonProperty("redblack")
+      RedBlack,
+
+      @JsonProperty("highlander")
+      Highlander,
 
-      public Options(
-          @JsonProperty("enableTraffic") Boolean enableTraffic,
-          @JsonProperty("services") List<String> services,
-          @JsonProperty("strategy") String strategy) {
-        this.enableTraffic = Optional.ofNullable(enableTraffic).orElse(false);
-        this.services = Optional.ofNullable(services).orElse(Collections.emptyList());
-        this.strategy = ManifestStrategyType.fromKey(strategy);
-      }
+      @JsonProperty("none")
+      None
     }
   }
+
+  @Override
+  public String getRawManifest() {
+    return manifests == null ? null : manifests.stream().findFirst().orElse(null);
```

Comment:
Why are we only returning the first manifest here?  The inline editor supports multiple manifests separated by `---` and I believe this will cause us to deploy only the first.

Reply:
👍